### PR TITLE
Move logic for platform selection to .h file.  Refs #23

### PR DESCRIPTION
**Pull request by cnlohr**
Also fix compile for wave_player on Linux

### CODE REVIEW

File: CNFA.h [hunk 1]
```diff
@@ -86,6 +86,24 @@ void RegCNFADriver( int priority, const char * name, CNFAInitFn * fn );
...
+                     || defined(__WIN32__)   || defined(__CYGWIN__) \
+                     || defined(__MINGW32__) || defined(__MINGW64__) \
+                     || defined(__TOS_WIN__)
+#define CNFA_WINDOWS 1
+#elif defined( ANDROID ) || defined( __android__ ) || defined(ANDROID)
+#define CNFA_ANDROID 1
+#elif defined(__NetBSD__) || defined(__NetBSD) || defined(__sun) || defined(sun)
+#define CNFA_SUN 1
+#elif defined(__linux) || defined(__linux__) || defined(linux) || defined(__LINUX__)
+#define CNFA_ALSA 1
```
Comment (AEFeinstein) [inline CNFA.h]: This feels like it should be `CNFA_LINUX`

**Comment by sellicott** [comment]
This does look cleaner to me, I haven't tested for functionality on my system though.

**Comment by AEFeinstein** [comment]
I tested these changes on Windows, msys2, and they work fine. I did have to manually compile it though. `-lasound` fails and `-lwinmm` is required. Might be nice to update the makefile to be OS aware. You can see an example of that here:

https://github.com/AEFeinstein/Super-2024-Swadge-FW/blob/main/makefile

**Comment by cnlohr** [comment]
Added OS detection, I plan to merge in a few hours.

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**